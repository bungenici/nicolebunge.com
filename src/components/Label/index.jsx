import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class Label extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    htmlFor: PropTypes.string.isRequired,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children, htmlFor } = this.props;

    return (
      <label
        className={styles.label}
        htmlFor={htmlFor}
      >
        {children}
      </label>
    );
  }
}

export default Label;
