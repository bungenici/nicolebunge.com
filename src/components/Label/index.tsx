import classNames from 'classnames';
import React from 'react';
import styles from './style.module.css';

export interface LabelProps {
  className?: string;
  htmlFor: string;
}

const Label: React.FC<LabelProps> = (props) => {
  const { children, className, htmlFor, ...otherProps } = props;

  return (
    <label className={classNames(className, styles.label)} htmlFor={htmlFor} {...otherProps}>
      {children}
    </label>
  );
};

export default Label;
