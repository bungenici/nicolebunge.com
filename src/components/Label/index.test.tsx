/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Label from '.';

describe('Label component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Label htmlFor="name" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
