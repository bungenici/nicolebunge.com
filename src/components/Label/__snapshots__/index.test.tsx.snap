// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component renders correctly 1`] = `
<DocumentFragment>
  <label
    class="label"
    for="name"
  />
</DocumentFragment>
`;
